cpp/CMakeLists.txt

```diff
@@ -162,9 +162,6 @@ if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
     "Build with zstd compression"
     ON)
 
-  option(ARROW_ZLIB_VENDORED
-    "Build our own zlib (some libz.a aren't configured for static linking)"
-    ON)
```

Comment:
Did you intend to remove this? If we remove this, we should make sure that we don't search in the default system path on Linux, because the `libz.a` on some Linux distributions is not compiled with -fPIC